data/test/scenarios/wml_tests/FilterWML/special_id_active.cfg

```diff
@@ -0,0 +1,111 @@
+# This unit test reproduces setup of weapon special such as https://github.com/inferno8/wesnoth-Era_of_Magic/blob/adf227cba2e66602c4bb49dc203fa751865aa124/utils/beam.cfg, which is not supposed to work in rpg.
+# Focus of this test is whether [filter_attack]special_id_active work for [filter_self]lua_function controlled conditional weapon special
+
+{GENERIC_UNIT_TEST "special_id_active_lua_function" (
+    [event]
+        name=preload
+        first_time_only=no
+        [lua]
+            code=<<
+conditional_special_active = function()
+	return wml.variables.conditional_special_active
+end
+>>
+        [/lua]
+    [/event]
+    [event]
+        name=attacker_hits,attacker_misses
+        first_time_only=no
+
+        [filter_attack]
+            special_id_active=conditional_special
+        [/filter_attack]
+        {VARIABLE_OP events_trigger_result add 1}
+    [/event]
+    [event]
+        name=run_test_case
+        first_time_only=no
+        [modify_unit]
+            [filter]
+            [/filter]
+            max_hitpoints=100
+            hitpoints=100
+            attacks_left=1
+        [/modify_unit]
+        [store_unit]
+            [filter]
+                id=alice
+            [/filter]
+            variable=a
+            kill=yes
+        [/store_unit]
+        [store_unit]
+            [filter]
+                id=bob
+            [/filter]
+            variable=b
+        [/store_unit]
+        [unstore_unit]
+            variable=a
+            find_vacant=yes
+            x,y=$b.x,$b.y
```

Comment:
Instead of `GENERIC_UNIT_TEST`, using `COMMON_KEEP_A_B_UNIT_TEST` will put Alice in position to attack, and then you can remove this `[unstore_unit]` along with the `[store_unit]` on line 35.

Reply:
done